activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1620,13 +1620,17 @@ def build_where_clause(opts, rest = []) # :nodoc:
         case opts
         when String
           if rest.empty?
-            parts = [Arel.sql(opts)]
+            if Arel.arel_node?(opts)
+              parts = [opts]
+            else
+              parts = [Arel.sql(opts)]
+            end
           elsif rest.first.is_a?(Hash) && /:\w+/.match?(opts)
             parts = [build_named_bound_sql_literal(opts, rest.first)]
           elsif opts.include?("?")
             parts = [build_bound_sql_literal(opts, rest)]
           else
-            parts = [model.sanitize_sql(rest.empty? ? opts : [opts, *rest])]
+            parts = [Arel.sql(model.sanitize_sql([opts, *rest]))]
```

Comment:
The `rest.empty?` check appears to have been dead code since 8e6a5deca626c944933211d7977f4e7e76869e74